File: pe-core/src/main/java/org/saar/core/screen/Screen.java
package org.saar.core.screen;

import org.saar.lwjgl.opengl.fbos.FboBlitFilter;
import org.saar.lwjgl.opengl.utils.GlBuffer;

public interface Screen {

    int getWidth();

    int getHeight();

    default void copyTo(Screen other) {
        copyTo(other, FboBlitFilter.LINEAR, GlBuffer.COLOUR);
    }

    void copyTo(Screen other, FboBlitFilter filter, GlBuffer... buffers);

    void setAsDraw();

    void setAsRead();

}

File: pe-core/src/main/java/org/saar/core/screen/Screen.kt
package org.saar.core.screen

import org.saar.lwjgl.opengl.fbo.FboBlitFilter
import org.saar.lwjgl.opengl.utils.GlBuffer
import org.saar.lwjgl.opengl.utils.GlUtils

interface Screen {

    val width: Int

    val height: Int

    fun copyTo(other: Screen,
               filter: FboBlitFilter = FboBlitFilter.LINEAR,
               vararg buffers: GlBuffer = arrayOf(GlBuffer.COLOR))

    fun setAsDraw()
}

fun Screen.copyTo(other: Screen) = copyTo(other, FboBlitFilter.LINEAR, GlBuffer.COLOR)

fun Screen.clear(vararg buffers: GlBuffer) = this.clear(buffers.asIterable())

fun Screen.clear(buffers: Iterable<GlBuffer>) {
    this.setAsDraw()
    GlUtils.clear(buffers)
}